### DISCUSSION TITLE
Anisotropic porous friction?

### DISCUSSION THREAD

**Discussion by veeshy** [idaholab/moose#27001] [Q&A Modules: Navier-Stokes]
Anyone have thoughts on how to implement an anisotropic porous friction factor in PINSFV to model a perforated plate like in https://www.simscale.com/knowledge-base/how-to-use-porous-media-feature-to-define-perforated-plate/ ? For flow aligned to an axis it seems easy enough, but I'm not sure what to do if the plate is angled like in the linked example.

**Comment by GiudGiud** [comment]
Hello

I would create a new friction material. In this material, make the friction coefficient dependent on the direction d of the perforations and the velocity. Maybe just the scalar product d * v.
if d and v are alight, K is minimum, as d * v decreases, K increases to block flow

Guillaume

**Reply by veeshy** [reply]
Hm, I don't think just the scalar product will work because it won't artificially inflate the transverse friction. For example if the perforation direction was {1, 0, 0}, I'd want my friction to be {F, 10*F, 10*F}. And I don't know what the equivalent is if the direction were {0.5, 0.5 0}

**Reply by GiudGiud** [reply]
pretty sure it can work. Pseudo code:
if (d * v > 0.9 * norm(d) * norm(v))
  F_x = F_y = F_z = 1
else
  F_x = F_y = F_z = 10

this penalizes other directions than d.
You ll want something smooth instead of this to get better convergence

**Reply by veeshy** [reply]
oh, yeah that does look better than what I though. Seems to generally work on its way to convergence then diverges. I'll think about a smoother way to get there. Thanks for the suggestion!

**Reply by veeshy** [reply]
I tried this out, I'm not sure it's quite right. I was able to get some mild convergence but never quite there. I have a pipe with a directional perforated plate in it not aligned to an axis. I rotated the mesh so that the plate is normal to the x axis. I expect both cases to give the same results and that the x axis case should simply be the case where I can input the x direction loss factor and keep the y direction much larger. 

I don't think the v dotted with the plate orientation can capture the "actual" velocity where the flow isn't actually in the perforation direction where the wall would be blocking the perfection exit. 
<img width="424" alt="image" src="https://github.com/idaholab/moose/assets/5826161/28e4bb58-816c-4a75-b442-5728ac08aecd">

I tried a case of rotating the loss factors from the perforation axis to the global one, that ends up having a much higher pressure drop.

**Reply by veeshy** [reply]
Ah, I think simply rotating the friction vector isn't right because I'm using the action version of PINSFV which assumes the loss term only depends on the principle directions. I think a shear-like term needs to be there with a tensor form of the loss coefficient * the velocity vector (or something like that..)
